code/script.js

```diff
@@ -220,106 +227,177 @@ const generateBoard = () => {
+    const filterCharacters = (keep) => {
+      console.log(keep);
+      const { category, value } = currentQuestion
+      // Show the correct alert message for different categories
+      if (category === 'accessories') {
+        if (keep) {
+          alert(
+            `Yes, the person wears ${value}! Keep all people that wears ${value}`
+          )
+        } else {
+          alert(
+            `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+          );
+        }
+      } else if (category === 'other') {
+        alert(
+          `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        );
+      } else {
+        if (keep) {
+          alert(
+            `Yes, the person wears ${value}! Keep all people that wears ${value}`
+          );
+        } else {
+          alert(
+            `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+          );
+        }
+    else {
+  if (keep) {
+    alert(
+      `Yes, the person has ${value} ${category}! Keep all people with ${value} ${category}`
+    );
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
-    }
+    alert(
+      `No,the person doesn't have ${value} ${category}! Remove all people with ${value} ${category}`
+    );
   }
+}
+
+// Determine what is the category
+// filter by category to keep or remove based on the keep variable.
+
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+if (category === "hair" || category === "eyes") {
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] === value
+    );
+  } else {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => person[category] !== value
+    );
+  }
+} else if (category === "accessories" || category === "other") {
+  if (keep) {
+    charactersInPlay = charactersInPlay.filter((person) =>
+      person[category].includes(value)
+    );
+  } else {
+    charactersInPlay = charactersInPlay.filter(
+      (person) => !person[category].includes(value)
+    );
+  }
 }
+// Invoke a function to redraw the board with the remaining people.
+generateBoard(keep);}
```

Comment:
I am guessing that your board isn't generating at the startup of the game when this function is called because "keep" doesn't have a value. I could be wrong as I can't get to that part in the code when I run it but when I try following the logic I don't see where the variable "keep" is ever actually assigned a value.